src/toil/test/cwl/cwlTest.py

```diff
@@ -631,6 +639,162 @@ def test_kubernetes_cwl_conformance(self, **kwargs):
+
+            # Also collect its output
+            out_data = p.stdout.read()
+            if out_data:
+                out_buffer += out_data
+
+            while out_buffer.find(b'\n') != -1:
+                # And log every full line
+                cut = out_buffer.find(b'\n')
+                log.info('STDOUT: %s', out_buffer[0:cut].decode('utf-8', errors='ignore'))
+                loops_since_line = 0
+                out_buffer = out_buffer[cut+1:]
+
+            # Same for the error
+            err_data = p.stderr.read()
+            if err_data:
+                err_buffer += err_data
+
+            while err_buffer.find(b'\n') != -1:
+                cut = err_buffer.find(b'\n')
+                log.info('STDERR: %s', err_buffer[0:cut].decode('utf-8', errors='ignore'))
+                loops_since_line = 0
+                err_buffer = err_buffer[cut+1:]
+
+            loops_since_line += 1
+            if loops_since_line > 60:
+                log.debug('...waiting...')
+                loops_since_line = 0
+
+            time.sleep(1)
+
+        # At the end, log the last lines
+        if out_buffer:
+            log.info('STDOUT: %s', out_buffer.decode('utf-8', errors='ignore'))
+        if err_buffer:
+            log.info('STDERR: %s', err_buffer.decode('utf-8', errors='ignore'))
+
+        if p.returncode != 0:
+            # It failed
+            log.error("Failed to run %s.", str(cmd))
+            raise subprocess.CalledProcessError(p.returncode, ' '.join(cmd))
+
+    def createClusterUtil(self, args=None):
+        args = [] if args is None else args
+
+        command = ['toil', 'launch-cluster', '-p=aws', '-z', self.zone, f'--keyPairName={self.keyName}',
+                   '--leaderNodeType=t4g.medium', '-T', 'kubernetes', '--logDebug', self.clusterName] + args
+
+        log.debug('Launching cluster: %s', command)
+
+        # Try creating the cluster
+        subprocess.check_call(command)
+        # If we fail, tearDown will destroy the cluster.
+
+    def launchCluster(self):
+        self.createClusterUtil()
+
+    def testCWLOnARM(self, preemptableJobs=False):
+        """Does the work of the testing.  Many features' tests are thrown in here in no particular order."""
+        # Make a cluster
+        self.launchCluster()
+        # get the leader so we know the IP address - we don't need to wait since create cluster
+        # already insures the leader is running
+        self.cluster = cluster_factory(provisioner='aws', zone=self.zone, clusterName=self.clusterName)
+        self.leader = self.cluster.getLeader()
+
+        # Clone Toil repo
+        # We retrieve the Toil version needed to run the CWL tests
+        # Use either a git command, CI_COMMIT_SHA, or with version.py to find the version
+        # After finding the correct commit and checkout
+        # Activate the virtualenv
+        # In the commit we run the tests using pytest
+
+        commit = os.environ['CI_COMMIT_SHA']
```

Comment:
If the test needs this variable set, we should put a marker on it to require that.

I think we should bring back the `needs_env_var` decorator from https://github.com/DataBiosphere/toil/pull/3956/files/f1b1b854d9f8bd288ed51252884043b6cf176173#diff-3425fd4766c61da9e32e2b8f70b5d5dece213fd8ceed813bf8370034913f7d28R325-R336 and put `@needs_env_var('CI_COMMIT_SHA')` on this test. Otherwise it will fail instead of being skipped if someone tries to run it locally and doesn't fake the variable.